### ISSUE TITLE
Hosted VS2017 update to 2.0 breaking build

### ISSUE THREAD

**Issue by marcwmiller** [Microsoft/vsts-agent#1155]
A previously working build (for several months) started failing after Monday, August 21, 2017 3:00 PM (EDT). 

The CLI command dotnet restore was exiting with:

##[error]Process completed with exit code 1.

This was caused by:

...error NU1602: Twilio 5.6.2 does not provide an inclusive lower bound for dependency...

I enabled verbosity diag, and discovered that it is now using the 2.0 sdk files:

C:\Program Files\dotnet\sdk\2.0.0\MSBuild.dll /NoLogo /m /t:Restore /v:m /verbosity:diag .\iOnboard.sln

When I explicitly referenced C:\Program Files\dotnet\sdk\1.0.4\MSBuild.dll in the dotnet arguments, the nuget restore worked again.

Next I'm finding the Visual Studio solution is no longer building, once the dotnet restore workaround is in place.

From the build solution step I'm getting:

C:\Program Files\dotnet\sdk\2.0.0\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.Sdk.DefaultItems.targets (286, 5)
C:\Program Files\dotnet\sdk\2.0.0\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.Sdk.DefaultItems.targets(286,5): Error : Duplicate 'Content' items were included. The .NET SDK includes 'Content' items from your project directory by default. You can either remove these items from your project file, or set the 'EnableDefaultContentItems' property to 'false' if you want to explicitly include them in your project file. For more information, see https://aka.ms/sdkimplicititems. The duplicate items were: 'wwwroot\images\iologo.png'

How do I specify that I don't want to use dotnet sdk 2.0.0 on the Hosted VS2017 agent, since it is causing errors? I need to use dotnet sdk 1.0.4, which was working fine prior to today.

**Comment by bryanmacfarlane** [comment]
@marcwmiller - the hosted vs2017 pool contains an image with the latest released tools from vs 2017 so net core 1.0.4 was updated to 2.0 today.

If you want to lock to a specific version, you have a couple of options:
1. Use the .net tool installer task and specify the version you want to use (in the tools category) or
2. Use a private agent where you can control the tools installed.

Hope that helps.  

I'll keep this open as we go through your issue.

**Comment by bryanmacfarlane** [comment]
Does it work if you use the nuget tool installer task up front and select 4.1?

**Comment by bryanmacfarlane** [comment]
Can you open an issue on the vsts-tasks repo?  The agent in this repo has no concept of net core etc...  It's just an app.

**Comment by bryanmacfarlane** [comment]
@marcwmiller - I see you didn't open a case there.  Is your issue resolved?

**Comment by marcwmiller** [comment]
@bryanmacfarlane - sorry, I got pulled into another issue. I implemented the work around you suggested to use the .NET Core tool installer task. That resolved the issue. I will need to upgrade the local dev environment with VS 2017 v15.3 and dotnet 2.0 sdk to see if I can repro issue I encountered. Thank you for the follow up.

**Comment by bryanmacfarlane** [comment]
OK.  You mean you used it to pin yourself back to 1.1?  If so, would be good to know what broke with 2.0. :)

**Comment by marcwmiller** [comment]
Yes - I used it to install .NET Core sdk version 1.0.4, prior to running command dotnet restore.

Without that task in the build process, the command running dotnet restore was exiting with:

##[error]Process completed with exit code 1.

I *think* this was caused by a 3rd party nuget package:

2017-08-21T21:21:44.2532054Z d:\a\3\s\src\iOnboard.Service.Twilio\iOnboard.Service.Twilio.csproj : error NU1602: Twilio 5.6.1 does not provide an inclusive lower bound for dependency Newtonsoft.Json (> 9.0.1). An approximate best match of Newtonsoft.Json 10.0.1 was resolved. [d:\a\3\s\iOnboard.sln]

I should note that in all my .csproj files I have this:

  <PropertyGroup>
    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
  </PropertyGroup>

Possibly, moving from 1.0.4 to 2.0.0 with MSBuild.dll, it is now using that flag when doing restore, where it wasn't prior to 2.0.0.

**Comment by marcwmiller** [comment]
I updated the .csproj files, which have the 3rd party nuget package (Twilio) that generates the NU1602 warning, to explicitly suppress that warning and removed the .NET Core tool installer task from the build process. That resolved the issue. So again, I think moving from 1.0.4 -> 2.0.0 the restore is using the flag to treat warning as errors. I don't really think this is a bug.

**Comment by TingluoHuang** [comment]
I am going to close the issue as it's not an agent bug and you got unblocked. :)
